Source/Schema.NET/ValuesConverter.cs

```diff
@@ -74,5 +207,28 @@ public virtual void WriteObject(JsonWriter writer, object value, JsonSerializer
             var token = JToken.FromObject(value, serializer);
             token.WriteTo(writer);
         }
+
+        private static Type UnwrapNullable(Type type)
+        {
+            return type.GetTypeInfo().IsGenericType && type.GetGenericTypeDefinition() == typeof(Nullable<>)
+                ? Nullable.GetUnderlyingType(type)
+                : type;
+        }
+
+        private static bool IsPrimitiveType(Type type)
+        {
+            return type.GetTypeInfo().IsPrimitive || type == typeof(string);
+        }
+
+        private static object SanitizeReaderValue(JsonReader reader, JsonToken tokenType)
+        {
+            return tokenType == JsonToken.Integer ? Convert.ToInt32(reader.Value) : reader.Value;
```

Comment:
Integers will always come as Int64. We probably need a more resilient way to handle, but have not figured it out yet.